src/algorithms/base_algorithm.h

```diff
@@ -96,6 +98,8 @@ class BaseAlgorithm {
   }
 
  public:
+  virtual bool requires_conjugate_hierarchy() = 0;
```

Comment:
What's the purpose of this function? We already check this in `initialize()`

Reply:
I used it for testing, also I think it's nice to have a way of accessing this at runtime

Reply:
Fair enough. Then please move the check in `Neal2Algorithm::initialize()` to the `BaseAlgorithm` version